src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Selectors/X509CertificateValidator.cs

```diff
@@ -112,9 +111,14 @@ static bool StoreContainsCertificate(StoreName storeName, X509Certificate2 certi
                 try
                 {
                     store.Open(OpenFlags.ReadOnly);
-                    certificates = store.Certificates.Find(X509FindType.FindByThumbprint, certificate.GetCertHash(), false);
+                    certificates = store.Certificates.Find(X509FindType.FindByThumbprint, certificate.Thumbprint, false);
                     return certificates.Count > 0;
                 }
+                catch(Exception e)
```

Comment:
Remove exception handling here. The call path to this method is expecting an exception on validation failure.